cfpb/consumerfinance.gov
# Deprecate use of AbstractFilterPage preview fields

This commit deprecates the use of 4 preview-related fields on the AbstractFilterPage base page class: `preview_title`, `preview_subheading`, `preview_description`, and `preview_image`.

Previously on both regular post previews ([example](https://www.consumerfinance.gov/about-us/blog/)) and condensed post previews ([example](https://www.consumerfinance.gov/activity-log/)), the template would use `preview_title` if set, otherwise it would use `title`. With this change, these templates use Wagtail's built-in `seo_title` field if set, otherwise it will fall back to `title`.

Regular post previews now use `search_description` instead of `preview_description` for the post's description text. Regular post previews no longer render the `preview_subheading` field if defined.

Regular post previews no longer render the `preview_image` field; this will impact how some page results appear, [for example](https://www.consumerfinance.gov/about-us/blog/?title=year+you+buy+your+first+home&from_date=&to_date=).

Previously, the Elasticsearch-based search of AFPs indexed the `preview_title`, `preview_subheading`, and `preview_description` field as part of its index, and included those fields in its search. It no longer does. Search still includes both a page's title field and its content field.

Filterable list RSS feeds ([example](https://www.consumerfinance.gov/about-us/blog/feed/)) now use `search_description` instead of `preview_description` to populate the `<description>` field in the feed XML.

Note that the preview fields mentioned above still exist in the database and no data will be lost with this PR. They have been removed from the Wagtail admin edit view and will no longer be visible to editors. A future PR will remove the fields completely.

This PR also deprecates the Page Preview Fields report previously available in Wagtail.

## How to test this PR

Compare the production [blog](https://www.consumerfinance.gov/about-us/blog/) against [localhost](http://localhost:8000/about-us/blog/). Note the change with posts that previously had preview images, [production](https://www.consumerfinance.gov/about-us/blog/year-you-buy-first-home/) vs. [local](http://localhost:8000/about-us/blog/year-you-buy-first-home/). Confirm that [local RSS feed](http://localhost:8000/about-us/blog/feed/) still works.


## Notes and todos

@csebianlander I note that some existing `search_description` fields don't use a closing period, for example [here](http://localhost:8000/about-us/blog/?page=2):

<img width="797" alt="image" src="https://github.com/cfpb/consumerfinance.gov/assets/654645/af5ddd97-e37c-40ef-ac15-75339c3d9ecf">

Is this something we should fix before this change goes out and, if so, is this something we can do via script?

## Checklist
